sklearn/metrics/pairwise.py

```diff
@@ -54,6 +55,94 @@ def _return_float_dtype(X, Y):
     return X, Y, dtype
 
 
+def flexible_concatenate(it, final_len=None):
```

Comment:
should probably be `_flexible_vstack` (private), or `sklearn.utils.flexible_vstack`. It only joins on `axis=0`